### PR TITLE
Use boms for grpc and opentelemetry in the examples

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-java/issues/1950

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Comment by jkwatson** [comment]
I just checked this out locally and it doesn't build. We *really* need to get the examples building as a part of CI.

**Review by carlosalberto — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@jkwatson are you sure? It works for me. Can you post what commands you use? I am able to run examples:

```bash
./gradlew fatJar
java -cp zipkin/build/libs/opentelemetry-examples-zipkin-all-0.1.0-SNAPSHOT.jar io.opentelemetry.example.ZipkinExample localhost 9411
```

**Comment by jkwatson** [comment]
> @jkwatson are you sure? It works for me. Can you post what commands you use? I am able to run examples:
> 
> ```shell
> ./gradlew fatJar
> java -cp zipkin/build/libs/opentelemetry-examples-zipkin-all-0.1.0-SNAPSHOT.jar io.opentelemetry.example.zipkin.ZipkinExample localhost 9411
> ```

Yeah, I just verified on a 2nd computer. the grpc example fails to generate the grpc stubs with this, but it works fine on the main branch. Try it with a `clean`.

**Comment by bogdandrutu** [comment]
Will fix later or tomorrow, and also going to add a GitHub action to check for this.

File: examples/build.gradle [hunk 1]
```diff
@@ -28,6 +28,11 @@ subprojects {
         }
     }
 
+    dependencies {
```
Comment (anuraaga) [inline examples/build.gradle:31]: I would add these in the projects, not top level, to reduce the drift for a user wanting to copy-paste this example out for their use.

**Comment by bogdandrutu** [comment]
@jkwatson merging this, will solve build and add action in #1962

**Pull request merged**